semantics/src/traversal/expressions.rs

```diff
@@ -157,10 +157,12 @@ fn expr_subscript(
                     // Index value is ignored. We may want to introduce a new location
                     // variant named StorageRuntime or something to suit this case.
                     Type::Map(_) => Location::Storage { index: 0 },
+                    Type::Tuple(_) => Location::Value,
                 }
             }
             Type::Array(_) => Location::Value,
             Type::Base(_) => unreachable!(),
+            Type::Tuple(_) => unimplemented!(),
```

Comment:
Again, I don't think it would be possible to index a tuple like the other sequence types, so I'd think this would be a semantic error. Also, I'm wondering why indexing a base type is marked as `unreachable`, I'd think this would throw a semantic error too. I realize I probably wrote this code :D

Reply:
Makes sense. I created `SemanticError::NotSubscriptable` and changed both arms.